### PR TITLE
Update to latest MLFlow and Python versions

### PR DESCRIPTION
**Pull request by ekzhang**
Hi there, thanks for making and maintaining this Docker image! I noticed that the MLFlow and Python versions are a bit out-of-date since the image was last updated, so this PR updates MLFlow in the image, giving access to all the latest features. 🙂 

I noticed you were taking PRs, so if you could accept this and push a new image tag, that would be really helpful!

### Changes

- Update MLFlow 1.3 -> MLFlow 1.13
- Change internal file system store to SQLite
  (file system URIs are unsupported for model artifact tracking stores)
- Update Python 3.7 -> Python 3.9
- Update boto3 1.9.240 -> boto3 1.16.46
- Remove some unused defaults in Dockerfile

### CODE REVIEW

**Comment by flmu** [comment]
@ekzhang Thanks for updating the versions!

File: mlflow_tracking_server/files/run.sh [hunk 1]
```diff
@@ -12,18 +12,8 @@ if [ -z $AWS_BUCKET ]; then
...
-
-if [ -z $AWS_SECRET_ACCESS_KEY ]; then
-  echo >&2 "AWS_SECRET_ACCESS_KEY must be set"
-  exit 1
-fi
-
 mkdir -p $FILE_DIR && mlflow server \
-    --backend-store-uri file://${FILE_DIR} \
+    --backend-store-uri sqlite:///${FILE_DIR}/sqlite.db \
     --default-artifact-root s3://${AWS_BUCKET}/artifacts \
```
Comment (flmu) [inline mlflow_tracking_server/files/run.sh:17]: Thank you very much for your work. I have only one small comment:
We need the relevant AWS envs (`AWS_BUCKET, AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY`) for our S3 bucket access. Could you please undo the deletion of the AWS envs?

**Review by flmu — CHANGES_REQUESTED** [review]

File: mlflow_tracking_server/files/run.sh (hunk 1 above)
Comment (ekzhang) [reply mlflow_tracking_server/files/run.sh:17]: @flmu I don't believe that I deleted the environment variables; you can still pass them in the same way as described in the README (actually, I'm passing them in that way as well). The only thing was that I deleted the undocumented defaults for the environment variables, which aren't valid anyway.
```dockerfile
ENV AWS_BUCKET bucket
ENV AWS_ACCESS_KEY_ID aws_id
ENV AWS_SECRET_ACCESS_KEY aws_key
```
Is this okay?

**Review by flmu — APPROVED** [review]

**Comment by ekzhang** [comment]
Thank you! 😀

**Pull request merged**